clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -768,8 +768,9 @@ public void abortTransaction() throws ProducerFencedException {
      * response after each one.
      * <p>
      * The result of the send is a {@link RecordMetadata} specifying the partition the record was sent to, the offset
-     * it was assigned and the timestamp of the record. If
-     * {@link org.apache.kafka.common.record.TimestampType#CREATE_TIME CreateTime} is used by the topic, the timestamp
+     * it was assigned and the timestamp of the record. If the producer is configured with acks = 0, the {@link RecordMetadata}
+     * will have offset = -1 because there is no wait for the acknowledge from the broker.
```

Comment:
`because there is no wait for the acknowledge from the broker` doesn't read very well. I wonder if we can just drop that part and just say `If the producer is configured with acks = 0, the {@link RecordMetadata} will have offset = -1.` WDYT?

Reply:
The sentence I added was for making it clear why the offset is -1. It could be clear for people who know Kafka but maybe not for beginners?

Reply:
So what about something like:
```
If the producer is configured with acks = 0, the {@link RecordMetadata} will have offset = -1 because the producer does not wait for the acknowledgement from the broker.
```